Sources/Client/Model/Channel/Channel.swift

```diff
@@ -297,23 +303,35 @@ extension Channel {
     /// A channel name.
     public var name: String? {
         get {
-            extraData?.name
+            if nameAndImageForCurrentUser.name == nil, let namingStrategy = namingStrategy {
+                // Save generated name to nameAndImageForCurrentUser since there isn't any
+                nameAndImageForCurrentUser.name = namingStrategy.name(for: User.current,
+                                                                      members: members.map(\.user).sorted(by: { $0.id > $1.id }))
```

Comment:
You can sort members inside `namingStrategy` by default. Why it's needed here and for imageURL again?